src/Symfony/Bundle/FrameworkBundle/KernelBrowser.php

```diff
@@ -46,7 +46,11 @@ public function __construct(KernelInterface $kernel, array $server = [], History
      */
     public function getContainer()
     {
-        return $this->kernel->getContainer();
+        if (null === ($container = $this->kernel->getContainer())) {
+            @trigger_error('Getting the container from a non-booted kernel is deprecated since 4.3.', E_USER_DEPRECATED);
```

Comment:
What about throwing here as the class will be released for the first time in 4.3? We would then only need to catch this exception in the legacy `Client` class and turn it into a deprecation.